workflows/ai_surgical_video/python/ai_surgical_video.py

```diff
@@ -651,7 +703,8 @@ def dynamic_flow_callback(op):
         elif self.source == "aja":
             self.add_flow(aja, source, {("video_buffer_output", "in")})
         else:
-            self.add_flow(replayer, source)
+            self.add_flow(replayer, frame_sampler)
+            self.add_flow(frame_sampler, source)
```

Comment:
Is there an advantage to having `frame_sampler` alter input to the overall application as opposed to only affecting output as seen by the recorder?

If the motivation is solely to support testing, I would suggest running the application with normal input and then sample select output frames for recording and validation, rather than altering the input data source as seen by the application.

I.e. replace with:

```
add_flow(recorder_format_converter, frame_sampler)
add_flow(frame_sampler, recorder)
```

Reply:
The update looks good, thanks!